src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,57 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            bool loaded = false;
+            try
+            {
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                serviceProvider.AddAssembliesToConfiguration(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
+                int extCount = 0;
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
```

Comment:
This doesn't seem right - every time someone calls load extension from ADS it's going to re-initialize all of the loaded extensions. I would think that you'd just want to call Initialize on the single assembly from AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) and then add it to the completionExtensions list. 

A big issue you have right now is this situation :

1. Extension assembly requested to be loaded. Let's say there's already 2 other ICompletionExtensionProvider extensions loaded as well
2. Calling Initialize on the new extension fails and it throws an exception - but the other two passed fine
3. We'll now call SendError with the error message as expected
4. But then we'll continue on and SendResult(true) since the other extensions loaded properly

Reply:
Make sense, I fixed it by calling Create to get a new ExtensionProvider to find the completion extensions in the passed in assembly.

Reply:
Hm, but now you're calling CreateAsync on this new provider. We still want the ServiceHostInstance.ServiceProvider to have all the assemblies initialized.

What I think you'll need to do is loop through and just skip any that are already initialized. 

---
In reply to: [304580100](https://github.com/microsoft/sqltoolsservice/pull/833#discussion_r304580100) [](ancestors = 304580100)

Reply:
Fixed, please review again.